nntrainer/layers/common_properties.h

```diff
@@ -1340,7 +1340,19 @@ class LoraRank : public PositiveIntegerProperty {
 public:
   static constexpr const char *key = "lora_rank"; /**< unique key to access */
   using prop_tag = uint_prop_tag;                 /**< property type */
-  ;
+};
+
+/**
+ * @brief LoRA scaling parameter
+ * @details It is used to set the scaling factor of LoRA, which is calculated as
+ * `scaling = alpha / rank` in the original paper. Defulat = 1.
+ */
+class LoraScaling : public Property<float> {
+public:
+  LoraScaling(float value = 1.0) : nntrainer::Property<float>(value) {}
+  static constexpr const char *key =
```

Comment:
Well, is ```const``` after ```constexpr``` meaningful?

Reply:
As I understood, `constexpr const char *` is equivalent to `const char* const`. In this code, it is conventionally used to make the key pointer not to be revised/replaced by another string. `const char *` is used to save an immutable string, and `constexpr` is used to make the pointer immutable. Please let me know if there’s something I missed. Thank you :)

Reply:
```constexpr``` implies ```const```, which means ```constexpr char *``` is equivalent to ```const char *```. ```constexpr char * const``` seems what you want.

Reply:
Thank you for your answer. I partially agree with your comment, and it is correct when it is applied to a general variable. However, in this case, I am applying the `constexpr` to the pointer, so it seems a little bit different. I think the `constexpr` is applied to the type (`const char*`) itself, not the value. (B.T.W. `const` right after `constexpr` is applied to the value) So, in this case, it seems okay to write `constexpr const char*`. Please refer to this link : https://www.reddit.com/r/cpp_questions/comments/15iwf22/static_constexpr_char_vs_static_constexpr_const/
Thnx!! :)